app/v2/controllers/V2MovementsController.scala

```diff
@@ -392,6 +392,24 @@ class V2MovementsControllerImpl @Inject() (
       _ <- routerService
         .send(messageType, request.eoriNumber, movementId, updateMovementResponse.messageId, source)
         .asPresentation
+        .leftMap {
+          err =>
+            updateSmallMessageStatus(
+              request.eoriNumber,
+              movementType,
+              movementId,
+              updateMovementResponse.messageId,
+              MessageStatus.Failed
+            )
+            err
+        }
+      result = updateSmallMessageStatus(
```

Comment:
no need for the `result = ` here as the result will be implied by the  ```
_ <- routerService
         .send
``` call. Also, if you want the value of something to be evaluated in a for-comprehension composed of EitherTs then you need to use <-  rather than =

Reply:
updated